operator/src/main/java/org/keycloak/operator/controllers/KeycloakUpdateJobDependentResource.java

```diff
@@ -0,0 +1,190 @@
+        builder.withName(name)
+                .withNamespace(keycloak.getMetadata().getNamespace())
+                .withLabels(Utils.allInstanceLabels(keycloak))
+                .withAnnotations(Map.of(KEYCLOAK_CR_HASH_ANNOTATION, keycloakHash(keycloak)));
+        return builder.build();
+    }
+
+    private static void addPodSpecTemplate(JobSpecFluent<?> builder, Keycloak keycloak, Context<Keycloak> context) {
+        var podTemplate = builder.withNewTemplate();
+        podTemplate.withMetadata(createMetadata(podName(keycloak), keycloak));
+        podTemplate.withSpec(createPodSpec(context));
+        podTemplate.endTemplate();
+    }
+
+    private static PodSpec createPodSpec(Context<Keycloak> context) {
+        var builder = new PodSpecBuilder();
+        builder.withRestartPolicy("Never");
+        addInitContainer(builder, context);
+        addContainer(builder, context);
+        builder.addNewVolume()
+                .withName(WORK_DIR_VOLUME_NAME)
+                .withNewEmptyDir()
+                .endEmptyDir()
+                .endVolume();
+        // For test KeycloakDeploymentTest#testDeploymentDurability
+        // it uses a pause image, which never ends.
+        // After 60 seconds, the job is terminated allowing the test to complete.
+        // TODO should be configurable?
+        builder.withActiveDeadlineSeconds(POD_DEADLINE_SECONDS);
+        return builder.build();
+    }
+
+    private static void addInitContainer(PodSpecBuilder builder, Context<Keycloak> context) {
+        var existing = CRDUtils.firstContainerOf(ContextUtils.getCurrentStatefulSet(context)).orElseThrow();
+        var containerBuilder = builder.addNewInitContainerLike(existing);
+        configureContainer(containerBuilder, INIT_CONTAINER_NAME, INIT_CONTAINER_ARGS);
+        containerBuilder.endInitContainer();
+    }
+
+    private static void addContainer(PodSpecBuilder builder, Context<Keycloak> context) {
+        var existing = CRDUtils.firstContainerOf(ContextUtils.getDesiredStatefulSet(context)).orElseThrow();
+        var containerBuilder = builder.addNewContainerLike(existing);
+        configureContainer(containerBuilder, CONTAINER_NAME, CONTAINER_ARGS);
+        containerBuilder.endContainer();
+    }
+
+    private static void configureContainer(ContainerFluent<?> containerBuilder, String name, List<String> args) {
+        containerBuilder.withName(name);
+        containerBuilder.withArgs(replaceStartWithUpdateCommand(containerBuilder.getArgs(), args));
+        // remove volumes, won't be used
+        containerBuilder.withVolumeDevices();
+        containerBuilder.withVolumeMounts();
+        // remove restart policy and probes
+        containerBuilder.withRestartPolicy(null);
+        containerBuilder.withReadinessProbe(null);
+        containerBuilder.withLivenessProbe(null);
+        containerBuilder.withStartupProbe(null);
+        // remove resource request/limit
+        containerBuilder.withResources(null);
```

Comment:
The pods will use memory, and using more memory on a node than available would lead to memory pressure, which is a dreaded situation on Kubernetes. To be on the safe side, I'd at least leave the memory settings as is. 
Providing a CPU request and limit could help to find a Node where there are enough resources available. 

So while we can be more cost-conscious than using the limits of the regular container, I'd say we're better off with the original settings than setting none at all. 

If people would need to change them, we could in the meantime offer them something like an unsupported `updateCheckJobTemplate`